esp/esp/program/modules/forms/admincore.py

```diff
@@ -44,3 +49,59 @@ def save_data(self):
     autocorrect = forms.BooleanField(initial=True, required=False, help_text="Check this box to attempt automatically adding lunch to a student's schedule so that they are less likely to violate the schedule constraint.")
     include_conditions = forms.BooleanField(initial=True, required=False, help_text="Check this box to allow students to schedule classes through lunch if they do not have morning or afternoon classes.")
 
+class ProgramSettingsForm(ProgramCreationForm):
+    """ Form for changing program-related settings. """
+    #Remove these fields because they are editable on the deadline management page
+    teacher_reg_start = None
+    teacher_reg_end   = None
+    student_reg_start = None
+    student_reg_end   = None
+    def __init__(self, *args, **kwargs):
+        super(ProgramSettingsForm, self).__init__(*args, **kwargs)
+
+    def save(self):
+        prog = self.instance
+        LineItemType.objects.filter(text='Program admission',program=prog
+        ).update(amount_dec=Decimal('%.2f' % self.cleaned_data['base_cost']))
+        prog.sibling_discount = self.cleaned_data['sibling_discount']
+        return super(ProgramSettingsForm, self).save()
```

Comment:
Why do we need this code in this form?  Shouldn't the `save` logic already be somewhere else?

Reply:
The `ProgramCreationForm`'s `save` only processes the model fields associated with the `Program` model, including the `name` and `url` fields. We then set the permission and accounting details after the fact in `commit_program`:
https://github.com/learning-unlimited/ESP-Website/blob/36158a4eb5a0214fa8c9612cbcc47c0436f014aa/esp/esp/program/views.py#L504-L506

https://github.com/learning-unlimited/ESP-Website/blob/32ed5e9816c99cdbe5f8555765e25b3d40cd2697/esp/esp/program/setup.py#L66